sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala

```diff
@@ -87,8 +101,8 @@ case class CreateHiveTableAsSelectCommand(
       }
     } else {
       try {
-        sparkSession.sessionState.executePlan(InsertIntoTable(
-          metastoreRelation, Map(), query, overwrite = true, ifNotExists = false)).toRdd
+        sparkSession.sessionState.executePlan(InsertIntoTable(metastoreRelation,
```

Comment:
here we only reorder the columns of the query plan by name, and construct a `InsertIntoTable` plan, other rules will take care of `InsertIntoTable` and do cast. But yes, we should add a test